module/mempool/stdmap/backdata/herocache/cache.go

```diff
@@ -0,0 +1,403 @@
+package herocache
+
+import (
+	"encoding/binary"
+	"time"
+	_ "unsafe" // for linking runtimeNano
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/mempool/stdmap/backdata/herocache/heropool"
+)
+
+//go:linkname runtimeNano runtime.nanotime
+func runtimeNano() int64
+
+const (
+	slotsPerBucket = uint64(16)
+
+	// slotAgeUnallocated defines an unallocated slot with zero age.
+	slotAgeUnallocated = uint64(0)
+
+	// telemetryCounterInterval is the number of required interactions with
+	// this back data prior to printing any log. This is done as a slow-down mechanism
+	// to avoid spamming logs upon read/write heavy operations. An interaction can be
+	// a read or write.
+	telemetryCounterInterval = uint64(1000)
+
+	// telemetryDurationInterval is the required elapsed duration interval
+	// prior to printing any log. This is done as a slow-down mechanism
+	// to avoid spamming logs upon read/write heavy operations.
+	telemetryDurationInterval = 10 * time.Second
+)
+
+// bIndex is data type representing a bucket index.
+type bIndex uint64
+
+// sIndex is data type representing a slot index in a bucket.
+type sIndex uint64
+
+// sha32of256 is a 32-bits prefix flow.Identifier used to determine the bIndex of the entity
+// it represents.
+type sha32of256 uint32
+
+// slot is an internal notion corresponding to the identifier of an entity that is
+// meant to be stored in this Cache.
+type slot struct {
+	slotAge     uint64          // age of this slot.
+	entityIndex heropool.EIndex // link to actual entity.
+	slotId      sha32of256      // slot id is the 32-bits prefix of entity identifier.
```

Comment:
Using `sha32of256` addresses comment [1], but why use `slotId` instead of e.g. the suggested `keySha32of256`?

`slotId` was original called `idPref` which was potentially confusing because some people read 'pref' as 'preference'. `slotId` also seems potentially confusing / misleading because if anything the ID is to do with the key (AKA entity) and not with the slot, or?

If the preference is not to use the word 'key' then why not call it `entitySha32of256` ?

Also, there are numerous places below where the original `idPref` is used still but not as part of this struct, e.g.:

`c.buckets[bucketIndex][slotToUse].slotId = idPref`

This is potentially more confusing because something with the confusing and different name is being assigned to 'slotId'. As a suggestion, why not also refactor away all `idPref` references whether in this struct or elsewhere. Then the example above would become e.g.:

`c.buckets[bucketIndex][slotToUse].entitySha32of256 = entitySha32of256`

Another possibility -- since the terminology for entity IDs is already established -- would be to refer to the ID instead, e.g.:

`c.buckets[bucketIndex][slotToUse].entityId32of256 = entityId32of256`

[1] https://github.com/onflow/flow-go/pull/1778#discussion_r779781043

Reply:
https://github.com/onflow/flow-go/pull/1778/commits/f656e316a87dc4f4c79fa7b01fce0cab1f746de3